extensions/calDotCom/schema.ts

```diff
@@ -0,0 +1,47 @@
+import { z } from 'zod'
+
+const SettingsSchema = z.object({
+  apiKey: z.string().nonempty('Missing API key'),
+})
+
+const GetBookingFieldsSchema = z.object({
+  bookingId: z.string().nonempty('Missing bookingId'),
+})
+
+export const GetBookingPayloadSchema = z.object({
+  fields: GetBookingFieldsSchema,
+  settings: SettingsSchema,
+})
+//
+
+// Leaving this and part of booking schema commented out, just in case we want to use it
+// it is a part of api response but we don't use it right now
+// const UserSchema = z.object({
+//   email: z.string(),
+//   name: z.string(),
+//   timeZone: z.string(),
+//   locale: z.string().optional().nullable(),
+// })
+
+// export type User = z.infer<typeof UserSchema>
+
+export const BookingSchema = z.object({
```

Comment:
@nckhell do you see any problem with this schema validation for getBooking response, would we ever allow data points to be saved in orchestration with invalid/missing values?

Reply:
Action will fail with validation error subactivity added. So Stef/Aisha will see it in the activity feed. Only now it will be clearer error that what we have now added by orchestration.

Reply:
Okay 👌